core/src/main/java/org/eclipse/dash/licenses/cli/GoSumFileReader.java

```diff
@@ -0,0 +1,241 @@
+package org.eclipse.dash.licenses.cli;
+
+import jakarta.json.JsonArray;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonValue;
+import org.eclipse.dash.licenses.ContentId;
+import org.eclipse.dash.licenses.IContentId;
+import org.eclipse.dash.licenses.cli.html_parser.JsoupProvider;
+import org.eclipse.dash.licenses.http.HttpClientService;
+import org.eclipse.dash.licenses.util.JsonUtils;
+import org.jsoup.nodes.Attribute;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.*;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class GoSumFileReader implements IDependencyListReader {
+
+  private final BufferedReader reader;
+  private final HttpClientService clientService;
+  private final JsoupProvider jsoupProvider;
+  private final String githubToken;
+  private final Map<String, String> commonHeaders = new HashMap<>();
+
+  private final int requestPageSize = 100;
+
+  public GoSumFileReader(InputStream input, HttpClientService clientService, JsoupProvider jsoupProvider) {
```

Comment:
For consistency, this should be reworked to use the injector rather than query the injector for values.